shared/src/main/scala/bloop/io/OsEnv.scala

```diff
@@ -0,0 +1,41 @@
+package bloop.io
+
+object OsEnv {
+  /*
+   * EOL is:
+   *    '\r\n' on Windows, unless SHELL is a unix-like shell environment
+   *    '\n'   everywhere else
+   */
+  lazy val EOL: String = eol
+
+  /*
+   * SHELL is defined in CYGWIN, MinGW, msys, Git Bash, etc.
+   */
+  def eol: String = getenvOpt("SHELL") match {
+    case Some(str) if str.matches(".*/(bash|csh|zsh|sh)$") =>
```

Comment:
What are potential values this matches? `/` looks like it should be escaped.

Maybe it would be simple instead of a regex to use:
```scala
val possibleShells =  Seq("bash" ,"csh","zsh","sh")
...
case Some(str) if possibleShells.exists(shell => str.endsWith(shell))
```

?

Reply:
I like your approach.   Here's the list of 18 shells in my /etc/shells:

```
/bin/sh
/bin/ash
/bin/bash
/bin/dash
/bin/mksh
/bin/pdksh
/bin/posh
/bin/tcsh
/bin/zsh
/usr/bin/sh
/usr/bin/ash
/usr/bin/bash
/usr/bin/dash
/usr/bin/mksh
/usr/bin/pdksh
/usr/bin/posh
/usr/bin/tcsh
/usr/bin/zsh
```
The common feature is that they all end is "sh", so maybe rather than iterate over a standard list, we can just assume that anytime SHELL is defined and ends with "sh", we can assume standard line endings.

Reply:
I think this should be fine. Unless there is a windows shell that works with the windows line endings?

Reply:
There are multiple workarounds for special cases, based on how the JVM is launched.
```
   unset SHELL                      # revert to default System.lineSeparator
   java -J-Dline.separator=$'\r\n'  # specify required value for lineSeparator
```

Reply:
BTW, the 2nd half of the entries in my `/etc/shells` are identical to the first half, but prefixed with `/usr`.

Since the code looks for an entry in `validShells` that is a substring of the actual `SHELL` setting,  `SHELL=/usr/bin/bash` will be matched with `/bin/bash`.